tests/prophet/test_prophet_model_export.py

```diff
@@ -0,0 +1,455 @@
+        return np.tile(
+            period, int(np.ceil(self.size / (self.seasonal_period / self.seasonal_freq)))
+        )[: self.size]
+
+    def _generate_raw(self):
+        base = np.random.lognormal(mean=2.0, sigma=0.92, size=self.size)
+        seasonal = [
+            np.polyval([-5.0, -1.0], x) for x in np.linspace(start=0, stop=2, num=self.size)
+        ]
+        series = (
+            np.linspace(start=45.0, stop=90.0, num=self.size) + base + seasonal + self._period_gen()
+        )
+        return series
+
+    def _generate_linear_data(self):
+        DataStruct = namedtuple("DataStruct", "dates, series")
+        series = self._generate_raw()
+        date_ranges = np.arange(
+            self.start, self.start + timedelta(days=self.size), timedelta(days=1)
+        ).astype(date)
+        return DataStruct(date_ranges, series)
+
+    def _generate_shift_data(self):
+        DataStruct = namedtuple("DataStruct", "dates, series")
+        raw = self._generate_raw()[: int(self.size * 0.6)]
+        temperature = np.concatenate((raw, raw / 2.0)).ravel()[: self.size]
+        date_ranges = np.arange(
+            self.start, self.start + timedelta(days=self.size), timedelta(days=1)
+        ).astype(date)
+        return DataStruct(date_ranges, temperature)
+
+    def _gen_series(self):
+        if self.shift:
+            return self._generate_shift_data()
+        else:
+            return self._generate_linear_data()
+
+    def create_series_df(self):
+        gen_data = self._gen_series()
+        temporal_df = pd.DataFrame.from_records(gen_data).T
+        temporal_df.columns = [self.date_field, self.target_field]
+        return temporal_df
+
+
+TEST_CONFIG = {
+    "shift": False,
+    "start": "2011-07-25",
+    "size": 365 * 4,
+    "seasonal_period": 7,
+    "seasonal_freq": 0.1,
+    "date_field": "ds",
+    "target_field": "y",
+}
+FORECAST_HORIZON = 60
+SEED = 98765
+HORIZON_FIELD_NAME = "horizon"
+TARGET_FIELD_NAME = "yhat"
+DS_FORMAT = "%Y-%m-%dT%H:%M:%S"
+INFER_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+ModelWithSource = namedtuple("ModelWithSource", ["model", "data"])
+
+
+@pytest.fixture(scope="session")
+def prophet_model():
+    np.random.seed(SEED)
+    data = DataGeneration(**TEST_CONFIG).create_series_df()
+    model = Prophet().fit(data)
+    return ModelWithSource(model, data)
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+@pytest.fixture
+def prophet_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(conda_env, additional_pip_deps=["pystan", "prophet", "pytest"])
+    return conda_env
+
+
+@pytest.fixture
+def future_horizon_df(model, horizon):
+    return model.make_future_dataframe(periods=horizon)
+
+
+@pytest.fixture
+def generate_forecast(model, horizon):
+    return model.predict(model.make_future_dataframe(periods=horizon))[TARGET_FIELD_NAME]
+
+
+@pytest.mark.large
```

Comment:
Can we add `pytestmark = pytest.mark.large` at the top of the file below the library imports, which will automatically mark all test cases as `pytest.mark.large` so that we don't have to do this for each test case & risk missing one?

Here's an example: https://github.com/mlflow/mlflow/blob/5f5fada1e1b423fefa5f99f6702377888b05f2a4/tests/pytorch/test_pytorch_autolog.py#L16

Reply:
Changed